config/v1/types_authentication.go

```diff
@@ -159,3 +170,74 @@ const (
 	// KubeConfigKey is the key for the kube config file data in a secret
 	KubeConfigKey = "kubeConfig"
 )
+
+type OIDCProvider struct {
```

Comment:
will we support required claims?

Reply:
I did not want to add that just yet and instead in case somebody needs it

Reply:
Added in the last patch